pkg/clients/roadmap_client/roadmap.go

```diff
@@ -0,0 +1,111 @@
+package roadmap_client
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/cache"
+	"github.com/content-services/content-sources-backend/pkg/config"
+	"github.com/redhatinsights/platform-go-middlewares/v2/identity"
+)
+
+type Meta struct {
+	Count int `json:"count"`
+	Total int `json:"total"`
+}
+
+type ModulesResponse struct {
+	ResponseMetadata Meta     `json:"meta"`
+	Data             []Module `json:"data"`
+}
+
+type Module struct {
+	ModuleName       string   `json:"module_name"`
+	RhelMajorVersion int      `json:"rhel_major_version"`
+	Streams          []Stream `json:"streams"`
+}
+
+type Stream struct {
+	StartDate string `json:"start_date"`
+	EndDate   string `json:"end_date"`
+	Version   string `json:"version"`
+	Stream    string `json:"stream"`
+}
+
+func encodedIdentity(ctx context.Context) (string, error) {
+	id := identity.GetIdentity(ctx)
+	jsonIdentity, err := json.Marshal(id)
+	if err != nil {
+		return "", fmt.Errorf("error marshaling json: %w", err)
+	}
+	return base64.StdEncoding.EncodeToString(jsonIdentity), nil
+}
+
+func (rc roadmapClient) GetAppstreams(ctx context.Context) (ModulesResponse, int, error) {
+	statusCode := http.StatusInternalServerError
+	server := config.Get().Clients.Roadmap.Server
+	var err error
+	var appStreamResp ModulesResponse
+	var body []byte
+
+	appstreams, err := rc.cache.GetRoadmapAppstreams(ctx)
+	if err != nil && !errors.Is(err, cache.NotFound) {
+		return ModulesResponse{}, statusCode, fmt.Errorf("error reading cache: %w", err)
+	}
+	if appstreams != nil {
+		err = json.Unmarshal(body, &appStreamResp)
```

Comment:
should this be `err = json.Unmarshal(appstreams, &appStreamResp)` instead?

seeing an unmarshal error when hitting the search endpoint more than once, i think maybe because the first call doesn't retrieve anything the cache

Reply:
also if we do retrieve something from the cache, i think we could probably return early here

Reply:
oh whoops, yeah. I also need to log an error when reading from the cache instead of returning